### PR TITLE
feat: add vm store cheatcode

### PR DESCRIPTION
**Pull request by Jrigada**
# What :computer: 
* Added vm.store cheatcode functionality

# Why :hand:
* downstream users of era-test-node would be able to use sthe store cheatcode to persist a value in a specific slot related to an address in storage. 

# Evidence :camera:
```
function testStore(bytes32 slot, bytes32 value) external {
    testStoreTarget testStoreInstance = new testStoreTarget();

    (bool success, ) = CHEATCODE_ADDRESS.call(
      abi.encodeWithSignature("store(address,bytes32,bytes32)", address(testStoreInstance), slot, value)
    );

    testStoreInstance.testStoredValue();
    require(success, "store failed");
  }
}

contract testStoreTarget {
  bytes32 public testValue = bytes32(uint256(0)); //slot 0

  function testStoredValue() public view {
    require(testValue == bytes32(uint256(115792089237316195423570985008687907853269984665640564039457584007913129639935)), "testValue was not stored correctly");
  }
}
```

Console output
```
  Cheatcodes
    ✔ Should test vm.store (328ms)


  1 passing (335ms)
```

# Notes  📝
When failing assert in function testStoredValue the era-test-node crashes with the following error
```
thread 'tokio-runtime-worker' panicked at 'assertion failed: `(left == right)`
  left: `115792089237316195423570985008687907853269984665640564039457584007913129639935`,
 right: `0`', /Users/jrigada/.cargo/git/checkouts/zksync-era-657998444dcd9ed9/71374c8/core/lib/multivm/src/versions/vm_refunds_enhancement/oracles/storage.rs:400:17
```

### CODE REVIEW

File: src/cheatcodes.rs [hunk 1]
```diff
@@ -270,6 +271,21 @@ impl<F: NodeCtx> CheatcodeTracer<F> {
                     u256_to_h256(pack_block_info(block_number, timestamp.as_u64())),
                 );
             }
+            Store(StoreCall {
+                account,
+                slot,
+                value,
+            }) => {
+                tracing::info!(
+                    "Setting storage slot {:?} for account {:?} to {:?}",
```
Comment (MexicanAce) [inline src/cheatcodes.rs]: nit: Should we prepend the 👷 emoji to match other CheatCodes and visibly show the user that something was manually changed from something other than a normal transaction?

File: e2e-tests/contracts/TestCheatcodes.sol [hunk 2]
```diff
@@ -53,4 +53,27 @@ contract TestCheatcodes {
...
+    testStoreInstance.testStoredValue();
+    require(success, "store failed");
+      
+  }
+}
+
+contract testStoreTarget {
+  bytes32 public testValue = bytes32(uint256(0)); //slot 0
+
+  function testStoredValue() public view {
```
Comment (MexicanAce) [inline e2e-tests/contracts/TestCheatcodes.sol]: Should we pass in the `value` here to validate that `testValue` is equal to it?

File: src/cheatcodes.rs (hunk 1 above)
Comment (Jrigada) [reply src/cheatcodes.rs]: Good Idea!

File: e2e-tests/contracts/TestCheatcodes.sol (hunk 2 above)
Comment (Jrigada) [reply e2e-tests/contracts/TestCheatcodes.sol]: Done!

**Pull request merged**